src/foam/lib/NetworkPropertyPredicate.js

```diff
@@ -0,0 +1,35 @@
+foam.CLASS({
+    package: 'foam.lib',
+    name: 'NetworkPropertyPredicate',
+    extends: 'foam.lib.PermissionedPropertyPredicate',
```

Comment:
Why is this extending a `PermissionedPropertyPredicate`? I'm not sure I see why they should be related.

Reply:
Will make PermissionedPropertyPredicate, NetworkPropertyPredicate and StoragePropertyPredicate independent implementations of the interface, and add an implementation that allows a combination of predicates.